src/flb_config_map.c

```diff
@@ -776,7 +776,7 @@ int flb_config_map_set(struct mk_list *properties, struct mk_list *map, void *co
                 *m_d_num = atof(kv->val);
             }
             else if (m->type == FLB_CONFIG_MAP_BOOL) {
-                m_bool = (char *) (base + m->offset);
+                m_bool = (int *) (base + m->offset);
```

Comment:
assigning the value 1 to the byte indexed at `base + m->offset` and read it combining with the 3 following bytes as int will lead to different result on big endian and little endian systems, 
on big endian systems, the 4 byte number |01|00|00|00| equals to 16777216(1<<24), on little endian systems it equals to 1.
This can be verified by the code snippet below:
```
#include <stdio.h>
#include <stdbool.h>

int main() {
    int a = 0;
    char *p = (char*)(&a);
    *p = 1;

    printf("The value of a is now: %d\n", a);

    bool b;
    char *p1 = (char*)(&b);
    *p1 = 1;
    printf("The value of b is now: %d\n", b);
    return 0;
}
```
The output on x86(little endian system):
```
The value of a is now: 1
The value of b is now: 1
```
on s390x(big endian system):
```
The value of a is now: 16777216
The value of b is now: 1
```

Reply:
Booleans are meant to be backed by integer sized storage in fluent-bit and this change is correct.